src/Voron/PageLocator.cs

```diff
@@ -18,55 +17,28 @@ private struct PageData
             public Page Page;
             [FieldOffset(16)]
             public bool IsWritable;
+            [FieldOffset(24)]
+            public int Generation;
```

Comment:
Assume that you have overflow, and hit the _old_ generation?

Reply:
No problem with old generation, because we are not counting on the generation being different. Doesn't matter the number itself as long as it is different.

Reply:
My fear is that the number would reset to being the same as the _current_ generation, and the page from the _old_ transaction would be returned.

Reply:
That should be addressed now.